### PR TITLE
Update pod template docs

### PR DESCRIPTION
**Pull request by anyasabo**
Relates to https://github.com/elastic/cloud-on-k8s/issues/1031

Adds docs for
* Pod template
* Custom plugins and bundles

and adds some links.

There is also a decent amount of overlap with the snapshots doc and the init containers for plugin install section, so I wonder if it is useful to have them in both or if we should centralize the plugin install description in the ES spec doc and then just reference it in the snapshots doc.

### CODE REVIEW

File: docs/elasticsearch-spec.asciidoc [hunk 1]
```diff
@@ -101,42 +144,94 @@ $ openssl req -x509 -newkey rsa:4096 -keyout tls.key -out tls.crt -days 365 -nod
...
+You can also refer to the Kubernetes documentation here for more information on pod templates:
+https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/#pod-templates[Pod Templates Overview]
+
+https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#podtemplatespec-v1-core[Pod Template Spec API Reference]
+
+
+[id="{p}-resource-requests-limits"]
+=== Resource Requests and Limits
+
+Resource requests and limits define how much RAM and CPU each individual container requires and how much they are allowed to use. You may want to set these in addition to the heap size as mentioned , though by default we will set the heap size to half of the container memory limit.
```
Comment (sebgl) [inline docs/elasticsearch-spec.asciidoc]: > we will set the heap size to half of the container memory limit.
we don't do that anymore.
Probably worth syncing with @barkbay about https://github.com/elastic/cloud-on-k8s/pull/1252/files. It looks like both PR overlap here.

File: docs/elasticsearch-spec.asciidoc [hunk 2]
```diff
@@ -101,42 +144,94 @@ $ openssl req -x509 -newkey rsa:4096 -keyout tls.key -out tls.crt -days 365 -nod
...
+        initContainers:
+        - name: install-plugins
+          image: docker.elastic.co/elasticsearch/elasticsearch:7.1.0"
+          command:
+          - sh
+          - -c
+          - |
+            bin/elasticsearch-plugin install --batch repository-gcs
+----
+
```
Comment (sebgl) [inline docs/elasticsearch-spec.asciidoc]: Would be nice here to document what init containers inherit by default: https://github.com/elastic/cloud-on-k8s/blob/master/operators/pkg/controller/common/defaults/pod_template.go#L229

File: docs/elasticsearch-spec.asciidoc [hunk 3]
```diff
@@ -146,7 +241,7 @@ The Elasticsearch cluster configuration can be updated at any time:
 On any change, ECK reconciles Kubernetes resources towards the desired cluster definition. Changes are done in a rolling fashion: the state of the cluster is continuously monitored, to allow addition of new nodes and removal of deprecated nodes.
 
 [id="{p}-change-budget"]
-==== Change budget
+=== Change budget
```
Comment (sebgl) [inline docs/elasticsearch-spec.asciidoc]: This was a subsection of `update-strategy`

**Comment by anyasabo** [comment]
Reworked a lot of the section organization. For resource requests, pointed to the separate compute doc. I wonder if it is worth keeping heap size and resource requests as individual sections in this doc, but then the content is just a link to the other doc to make it easier to find.

File: docs/elasticsearch-spec.asciidoc (hunk 1 above)
Comment (anyasabo) [reply docs/elasticsearch-spec.asciidoc]: Thanks, removed

File: docs/elasticsearch-spec.asciidoc [hunk 4]
```diff
@@ -3,24 +3,65 @@
...
+          # additional labels for pods
+          foo: bar
+----
+
+More common is setting resource requests and limits, which is covered in link:managing-compute-resources.asciidoc[Managing compute resources].
+
+You may want to also install additional plugins, which is described in the <<{p}-init-containers-plugin-downloads>> section.
+
+You may want to set environment variables to configure Elasticsearch. For instance to set the minimum and maximum JVM heap size to `2g` and `4g` respectively you may modify the environment variables of the `elasticsearch` container as follows:
+
```
Comment (thbkrkr) [inline docs/elasticsearch-spec.asciidoc]: Maybe we should merge #1348 first and refer to it here to show how to set env vars.

Comment (thbkrkr) [reply docs/elasticsearch-spec.asciidoc]: As Seb said, let's merge and fix overlaps in follow-up PRs.

File: docs/elasticsearch-spec.asciidoc [hunk 5]
```diff
@@ -116,6 +157,41 @@ spec:
         - dns: hulk.example.com
 ----
 
+[id="{p}-advanced-settings"]
+== Advanced Settings
+
+[id="{p}-virtual-memory"]
```
Comment (pebrc) [inline docs/elasticsearch-spec.asciidoc]: There is another `virtual-memory` anchor on this page on L289 that breaks the docs build for me on this PR

Comment (pebrc) [reply docs/elasticsearch-spec.asciidoc]:
```
INFO:build_docs:/out/html_docs/index.xml:1343: element section: validity error : ID k8s-virtual-memory already defined
INFO:build_docs:<section id="k8s-virtual-memory">
```
@artemnikitin any idea why this not flagged up in CI?

Comment (artemnikitin) [reply docs/elasticsearch-spec.asciidoc]: @pebrc based on CI logs everything was fine

Comment (artemnikitin) [reply docs/elasticsearch-spec.asciidoc]:
```
[2019-07-23T21:54:33.636Z] INFO:docker build:Successfully tagged docker.elastic.co/docs/build:1
[2019-07-23T21:54:33.636Z] INFO:build_docs:Building HTML from /doc/cloud-on-k8s/docs/index.asciidoc
[2019-07-23T21:54:38.927Z] INFO:build_docs:Done
```

Comment (pebrc) [reply docs/elasticsearch-spec.asciidoc]: What I am trying to say here is that it should fail. I raised https://github.com/elastic/cloud-on-k8s/issues/1392

**Pull request merged**